### PR TITLE
Addition of initial E2E tests

### PR DESCRIPTION
**Pull request by mpeterson**
There are several commits that are part of this. I recommend reviewing them individually to make it easier. 

When merged they should not be squashed to retain the history and granularity.

### CODE REVIEW

**Comment by mpeterson** [comment]
This is still a WIP but reviews are welcome and needed!

File: test/e2e/api/api.go [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+	AfterEach(func() {
+		federator = nil
+		testPod = nil
+		contexts = nil
+	})
+
+	testResourceAPIOperations := func(clusters ...string) {
+		By("Distributing")
+		err := federator.Distribute(testPod, clusters...)
+		Expect(err).ToNot(HaveOccurred())
```
Comment (mkolesnik) [inline test/e2e/api/api.go]: nit: You can expect it to succeed

File: test/e2e/api/api.go [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+		By("[slow] Checking correct distribution")
+		presence := framework.BuildPresenceMap(contexts, clusters...)
+		framework.AssertPodInClusters(f, testPod, presence)
+
+		By("Deleting")
+		err = federator.Delete(testPod)
+		Expect(err).ToNot(HaveOccurred())
```
Comment (mkolesnik) [inline test/e2e/api/api.go]: nit: You can expect it to succeed

File: test/e2e/api/api.go [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+	Context("when we want to distribute to all clusters", func() {
+		testResourceAPI()
+	})
+
+	Context("when we want to distribute to cluster2 only", func() {
+		testResourceAPI("cluster2")
+	})
+}
+
+func newPod(name, namespace string) *v1.Pod {
```
Comment (mkolesnik) [inline test/e2e/api/api.go]: We're also using newPod in resource_test, perhaps we can extract this to a shared test util class or something

**Comment by mkolesnik** [comment]
This is irrelevant as most or the code has changed substantially

**Pull request closed without merging**